cpp/cmake_modules/Findzstd.cmake

```diff
@@ -54,10 +54,11 @@ else()
   if(ZSTD_PC_FOUND)
     set(ZSTD_INCLUDE_DIR "${ZSTD_PC_INCLUDEDIR}")
 
-    if(NOT ARROW_ZSTD_USE_SHARED)
+    if(ARROW_ZSTD_USE_SHARED)
+      list(APPEND ZSTD_PC_LIBRARY_DIRS "${ZSTD_PC_LIBDIR}")
+    else()
```

Comment:
Some libzstd.pc versions are written in a way that only ${ZSTD_PC_STATIC_LIBDIR} contains a path. I started out with essentially this, but it didn't work during testing.

Reply:
Oh...
Could you show one of the libzstd.pc versions environment? I want to try it.

Reply:
It seems that `XXX_STATIC_LIBDIR` is never defined: https://github.com/Kitware/CMake/blob/master/Modules/FindPkgConfig.cmake#L566
`_pkg_get_variable` doesn't set `XXX_STATIC_YYY`: https://github.com/Kitware/CMake/blob/master/Modules/FindPkgConfig.cmake#L566
(`XXX_LIBDIR` is only defined.)